sound/soc/soc-core.c

```diff
@@ -2874,6 +2874,29 @@ static inline char *fmt_multiple_name(struct device *dev,
 	return kstrdup(dai_drv->name, GFP_KERNEL);
 }
 
+/**
+ * snd_soc_unregister_dai - Unregister DAI from the ASoC core
+ *
+ * @component: The component for which the DAI should be unregistered
+ * @dai_drv: the dai driver for the DAI
+ */
+void snd_soc_unregister_dai(struct snd_soc_component *component,
+			    struct snd_soc_dai_driver *dai_drv)
+{
+	struct snd_soc_dai *dai, *_dai;
+
+	for_each_component_dais_safe(component, dai, _dai) {
+		if (dai->driver == dai_drv) {
+			dev_dbg(component->dev, "ASoC: Unregistered DAI '%s'\n",
+				dai->name);
+			list_del(&dai->list);
+			kfree(dai->name);
+			kfree(dai);
+			return;
```

Comment:
this looks similar to snd_soc_unregister_dais(), except that you return here. would there be any merits in letting the loop continue?
Also the core is now quite confusing with (un)register_dais() and (un)register_dai() that are serve different purposes.